core/node/rpc/sync/client/syncer_set.go

```diff
@@ -37,12 +35,6 @@ type (
 		DebugDropStream(ctx context.Context, streamID StreamId) (bool, error)
 	}
 
-	// StreamCache represents a behavior of the stream cache
-	StreamCache interface {
-		GetStreamWaitForLocal(ctx context.Context, streamId StreamId) (*events.Stream, error)
-		GetStreamNoWait(ctx context.Context, streamId StreamId) (*events.Stream, error)
-	}
-
 	ModifyRequest struct {
```

Comment:
Add documentation to all callback in `ModifyRequest` that they must not lock a stream. If they lock a stream there is an opportunity for a deadlock.